Documentation/async-disaster-recovery.md

```diff
@@ -0,0 +1,460 @@
+    metadata:
+      name: rbd-volumereplicationclass
+    spec:
+      provisioner: rook-ceph.rbd.csi.ceph.com
+      parameters:
+        mirroringMode: snapshot
+        schedulingInterval: "12m"
+        schedulingStartTime: "16:18:43"
+        replication.storage.openshift.io/replication-secret-name: rook-csi-rbd-provisioner
+        replication.storage.openshift.io/replication-secret-namespace: rook-ceph
+    EOF
+    ```
+
+>:bulb: **Note:** The `schedulingInterval` can be specified in formats of
+> minutes, hours or days using suffix `m`,`h` and `d` respectively.
+> The optional schedulingStartTime can be specified using the ISO 8601
+> time format.
+
+* Once VolumeReplicationClass is created, create a Volume Replication for
+ the PVC which we intend to replicate to secondary cluster.
+
+    ```yaml
+    $cat <<EOF | kubectl --context=cluster-1 apply -f -
+    apiVersion: replication.storage.openshift.io/v1alpha1
+    kind: VolumeReplication
+    metadata:
+      name: pvc-volumereplication
+    spec:
+      volumeReplicationClass: rbd-volumereplicationclass
+      replicationState: primary
+      dataSource:
+        apiGroup: ""
+        kind: PersistentVolumeClaim
+        name: rbd-pvc # Name of the PVC on which mirroring is to be enabled.
+    EOF
+    ```
+
+>:memo: *VolumeReplication* is a namespace scoped object. Thus,
+> it should be created in the same namespace as of PVC.
+
+`replicationState` is the state of the volume being referenced.
+ Possible values are primary, secondary, and resync.
+
+* `primary` denotes that the volume is primary.
+* `secondary` denotes that the volume is secondary.
+* `resync` denotes that the volume needs to be resynced.
+
+To check VolumeReplication CR status:
+
+```bash
+kubectl get volumereplication pvc-volumereplication  --context=cluster-1 -oyaml
+```
+
+>```yaml
+>...
+>spec:
+>  dataSource:
+>    apiGroup: ""
+>    kind: PersistentVolumeClaim
+>    name: rbd-pvc
+>  replicationState: primary
+>  volumeReplicationClass: rbd-volumereplicationclass
+>status:
+>  conditions:
+>  - lastTransitionTime: "2021-05-04T07:39:00Z"
+>    message: ""
+>    observedGeneration: 1
+>    reason: Promoted
+>    status: "True"
+>    type: Completed
+>  - lastTransitionTime: "2021-05-04T07:39:00Z"
+>    message: ""
+>    observedGeneration: 1
+>    reason: Healthy
+>    status: "False"
+>    type: Degraded
+>  - lastTransitionTime: "2021-05-04T07:39:00Z"
+>    message: ""
+>    observedGeneration: 1
+>    reason: NotResyncing
+>    status: "False"
+>    type: Resyncing
+>  lastCompletionTime: "2021-05-04T07:39:00Z"
+>  lastStartTime: "2021-05-04T07:38:59Z"
+>  message: volume is marked primary
+>  observedGeneration: 1
+>  state: Primary
+>```
+
+* Take a backup of PVC and PV object on primary cluster(cluster-1)
+
+  * Take backup of the PVC `rbd-pvc`
+
+    ```bash
+    kubectl get pvc rbd-pvc -oyaml >pvc-backup.yaml
+    ```
+
+  * Take a backup of the PV, corresponding to the PVC
```

Comment:
Why do you need to take a backup of the PV?

Reply:
Since we create a static binding between the PVCs and the PVs on the secondary cluster, new PVs are not created there. The restored PV is then bound to the restored PVC.